src/utils/resolveDefaultProperties.ts

```diff
@@ -38,7 +38,7 @@ export function resolveDefaultProperties(sourceText: string) {
       return node;
     };
 
-    return (node) => ts.visitNode(node, visit);
+    return (node) => ts.visitNode(node, visit) as ts.SourceFile;
```

Comment:
I'm wondering why is it necessary now, I'm trying to use type assertion when it's the only way. This is bypassing typescript type checking so very dangerous (and from experience, where stuff break silently…)

Let's try to avoid this

Reply:
the definition of `visitNode` changed from typescript 4 to 5:

typescript 4:
```typescript
function visitNode<T extends Node>(node: T, visitor: Visitor | undefined, test?: (node: Node) => boolean, lift?: (node: readonly Node[]) => T): T;
```

typescript 5:
```typescript
    function visitNode<TIn extends Node | undefined, TVisited extends Node | undefined>(node: TIn, visitor: Visitor<NonNullable<TIn>, TVisited>, test?: (node: Node) => boolean, lift?: (node: readonly Node[]) => Node): Node | (TIn & undefined) | (TVisited & undefined);
```

so `undefined` was added as a possible return value and this causes an issue, even though we never return `undefined`.
so the type assertion SHOULD be fine, but I agree with you that it's not nice.